esp32 invalid flash dependencies

**Describe the bug**
The following was seen in a MCUmgr test which enables flash for the `icev_wireless` board:
```
FAILED: zephyr/zephyr_pre0.elf zephyr/zephyr_pre0.map 
: && ccache /opt/toolchains/zephyr-sdk-0.15.2/riscv64-zephyr-elf/bin/riscv64-zephyr-elf-gcc  -gdwarf-4 zephyr/CMakeFiles/zephyr_pre0.dir/misc/empty_file.c.obj -o zephyr/zephyr_pre0.elf  -fuse-ld=bfd  -Wl,-T  zephyr/linker_zephyr_pre0.cmd  -Wl,-Map=/__w/zephyr/zephyr/twister-out/icev_wireless/tests/subsys/mgmt/mcumgr/fs_mgmt_hash_supported/mgmt.mcumgr.fs.mgmt.hash.supported.crc32/zephyr/zephyr_pre0.map  -Wl,--whole-archive  app/libapp.a  zephyr/libzephyr.a  zephyr/arch/common/libarch__common.a  zephyr/arch/arch/riscv/core/libarch__riscv__core.a  zephyr/lib/libc/minimal/liblib__libc__minimal.a  zephyr/subsys/fs/libsubsys__fs.a  zephyr/subsys/mgmt/mcumgr/mgmt/libsubsys__mgmt__mcumgr__mgmt.a  zephyr/subsys/mgmt/mcumgr/smp/libsubsys__mgmt__mcumgr__smp.a  zephyr/subsys/mgmt/mcumgr/util/libsubsys__mgmt__mcumgr__util.a  zephyr/subsys/mgmt/mcumgr/grp/fs_mgmt/libsubsys__mgmt__mcumgr__grp__fs_mgmt.a  zephyr/subsys/mgmt/mcumgr/transport/libsubsys__mgmt__mcumgr__transport.a  zephyr/subsys/net/libsubsys__net.a  zephyr/subsys/testsuite/ztest/libsubsys__testsuite__ztest.a  zephyr/drivers/interrupt_controller/libdrivers__interrupt_controller.a  zephyr/drivers/clock_control/libdrivers__clock_control.a  zephyr/drivers/console/libdrivers__console.a  zephyr/drivers/gpio/libdrivers__gpio.a  zephyr/drivers/flash/libdrivers__flash.a  zephyr/drivers/serial/libdrivers__serial.a  zephyr/drivers/timer/libdrivers__timer.a  zephyr/drivers/pinctrl/libdrivers__pinctrl.a  modules/zcbor/libmodules__zcbor.a  -Wl,--no-whole-archive  zephyr/kernel/libkernel.a  zephyr/CMakeFiles/offsets.dir/./arch/riscv/core/offsets/offsets.c.obj  -L"/opt/toolchains/zephyr-sdk-0.15.2/riscv64-zephyr-elf/bin/../lib/gcc/riscv64-zephyr-elf/12.1.0"  -L/__w/zephyr/zephyr/twister-out/icev_wireless/tests/subsys/mgmt/mcumgr/fs_mgmt_hash_supported/mgmt.mcumgr.fs.mgmt.hash.supported.crc32/zephyr  -lgcc  zephyr/arch/common/libisr_tables.a  -lgcc  -no-pie  -mabi=ilp32  -march=rv32ima_zicsr_zifencei  -Wl,--gc-sections  -Wl,--build-id=none  -Wl,--sort-common=descending  -Wl,--sort-section=alignment  -Wl,-u,_OffsetAbsSyms  -Wl,-u,_ConfigAbsSyms  -nostdlib  -static  -Wl,-X  -Wl,-N  -Wl,--orphan-handling=warn  -Wl,--fatal-warnings  -T/__w/zephyr/modules/hal/espressif/zephyr/esp32c3/src/linker/esp32c3.rom.alias.ld  -T/__w/zephyr/modules/hal/espressif/zephyr/esp32c3/../../components/esp_rom/esp32c3/ld/esp32c3.rom.ld  -T/__w/zephyr/modules/hal/espressif/zephyr/esp32c3/../../components/esp_rom/esp32c3/ld/esp32c3.rom.eco3.ld  -T/__w/zephyr/modules/hal/espressif/zephyr/esp32c3/../../components/esp_rom/esp32c3/ld/esp32c3.rom.api.ld  -T/__w/zephyr/modules/hal/espressif/zephyr/esp32c3/../../components/esp_rom/esp32c3/ld/esp32c3.rom.libgcc.ld  -T/__w/zephyr/modules/hal/espressif/zephyr/esp32c3/../../components/soc/esp32c3/ld/esp32c3.peripherals.ld && cd /__w/zephyr/zephyr/twister-out/icev_wireless/tests/subsys/mgmt/mcumgr/fs_mgmt_hash_supported/mgmt.mcumgr.fs.mgmt.hash.supported.crc32/zephyr && /usr/local/bin/cmake -E echo
/opt/toolchains/zephyr-sdk-0.15.2/riscv64-zephyr-elf/bin/../lib/gcc/riscv64-zephyr-elf/12.1.0/../../../../riscv64-zephyr-elf/bin/ld.bfd: zephyr/libzephyr.a(flash_mmap.c.obj): in function `spi_flash_mmap_pages':
/__w/zephyr/modules/hal/espressif/components/spi_flash/flash_mmap.c:185: undefined reference to `k_malloc'
/opt/toolchains/zephyr-sdk-0.15.2/riscv64-zephyr-elf/bin/../lib/gcc/riscv64-zephyr-elf/12.1.0/../../../../riscv64-zephyr-elf/bin/ld.bfd: zephyr/libzephyr.a(flash_mmap.c.obj): in function `spi_flash_mmap':
/__w/zephyr/modules/hal/espressif/components/spi_flash/flash_mmap.c:151: undefined reference to `k_malloc'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

The Kconfig dependencies for this are therefore invalid

**Expected behavior**
Dependencies to be properly set

**Impact**
Showstopper, holding up and blocking completely unrelated PRs

> Yes, just tried it locally here with your patch and it builds fine

+1, I can confirm that this did the trick for me on esp32c3_devkitm, where spi_esp32_spim driver wouldn't work otherwise (undefined reference to k_calloc)

--- a/boards/riscv/esp32c3_devkitm/Kconfig.defconfig
+++ b/boards/riscv/esp32c3_devkitm/Kconfig.defconfig
@@ -7,13 +7,11 @@ config BOARD
 	default "esp32c3_devkitm"
 	depends on BOARD_ESP32C3_DEVKITM
 
-if BT
-
 config HEAP_MEM_POOL_SIZE
-	default 16384
+	default 98304 if WIFI
+	default 16384 if BT
+	default 4096
 
 choice BT_HCI_BUS_TYPE
-	default BT_ESP32
+	default BT_ESP32 if BT
 endchoice
-
-endif # BT

--- a/boards/riscv/icev_wireless/Kconfig.defconfig
+++ b/boards/riscv/icev_wireless/Kconfig.defconfig
@@ -5,13 +5,11 @@ config BOARD
 	default "icev_wireless"
 	depends on BOARD_ICEV_WIRELESS
 
-if BT
-
 config HEAP_MEM_POOL_SIZE
-	default 16384
+	default 98304 if WIFI
+	default 16384 if BT
+	default 4096
 
 choice BT_HCI_BUS_TYPE
-	default BT_ESP32
+	default BT_ESP32 if BT
 endchoice
-
-endif # BT

--- a/boards/riscv/xiao_esp32c3/Kconfig.defconfig
+++ b/boards/riscv/xiao_esp32c3/Kconfig.defconfig
@@ -10,10 +10,6 @@ config HEAP_MEM_POOL_SIZE
 	default 16384 if BT
 	default 4096
 
-if BT
-
 choice BT_HCI_BUS_TYPE
-	default BT_ESP32
+	default BT_ESP32 if BT
 endchoice
-
-endif # BT

--- a/boards/xtensa/esp32/Kconfig.defconfig
+++ b/boards/xtensa/esp32/Kconfig.defconfig
@@ -7,16 +7,14 @@ config BOARD
 	default "esp32"
 	depends on BOARD_ESP32
 
-if BT
-
-config HEAP_MEM_POOL_SIZE
-	default 16384
-
 config ENTROPY_GENERATOR
 	default y
 
-choice BT_HCI_BUS_TYPE
-	default BT_ESP32
-endchoice
+config HEAP_MEM_POOL_SIZE
+	default 98304 if WIFI
+	default 16384 if BT
+	default 4096
 
-endif # BT
+choice BT_HCI_BUS_TYPE
+	default BT_ESP32 if BT
+endchoice

--- a/boards/xtensa/esp32_ethernet_kit/Kconfig.defconfig
+++ b/boards/xtensa/esp32_ethernet_kit/Kconfig.defconfig
@@ -7,9 +7,6 @@ config BOARD
 	default "esp32_ethernet_kit"
 	depends on BOARD_ESP32_ETHERNET_KIT
 
-config ENTROPY_ESP32_RNG
-	default y if ENTROPY_GENERATOR
-
 config ESP_SPIRAM
 	default y
 
@@ -17,16 +14,14 @@ choice SPIRAM_TYPE
 	default SPIRAM_TYPE_ESPPSRAM64
 endchoice
 
-if BT
-
-config HEAP_MEM_POOL_SIZE
-	default 16384
-
 config ENTROPY_GENERATOR
 	default y
 
-choice BT_HCI_BUS_TYPE
-	default BT_ESP32
-endchoice
+config HEAP_MEM_POOL_SIZE
+	default 98304 if WIFI
+	default 16384 if BT
+	default 4096
 
-endif # BT
+choice BT_HCI_BUS_TYPE
+	default BT_ESP32 if BT
+endchoice

--- a/boards/xtensa/esp32_net/Kconfig.defconfig
+++ b/boards/xtensa/esp32_net/Kconfig.defconfig
@@ -7,19 +7,14 @@ config BOARD
 	default "esp32_net"
 	depends on BOARD_ESP32_NET
 
-config ENTROPY_ESP32_RNG
-	default y if ENTROPY_GENERATOR
-
-if BT
-
-config HEAP_MEM_POOL_SIZE
-	default 16384
-
 config ENTROPY_GENERATOR
 	default y
 
-choice BT_HCI_BUS_TYPE
-	default BT_ESP32
-endchoice
+config HEAP_MEM_POOL_SIZE
+	default 98304 if WIFI
+	default 16384 if BT
+	default 4096
 
-endif # BT
+choice BT_HCI_BUS_TYPE
+	default BT_ESP32 if BT
+endchoice

--- a/boards/xtensa/esp32s2_franzininho/Kconfig.defconfig
+++ b/boards/xtensa/esp32s2_franzininho/Kconfig.defconfig
@@ -9,3 +9,12 @@ config BOARD
 
 config ENTROPY_GENERATOR
 	default y
+
+config HEAP_MEM_POOL_SIZE
+	default 98304 if WIFI
+	default 16384 if BT
+	default 4096
+
+choice BT_HCI_BUS_TYPE
+	default BT_ESP32 if BT
+endchoice

--- a/boards/xtensa/esp32s2_saola/Kconfig.defconfig
+++ b/boards/xtensa/esp32s2_saola/Kconfig.defconfig
@@ -9,3 +9,12 @@ config BOARD
 
 config ENTROPY_GENERATOR
 	default y
+
+config HEAP_MEM_POOL_SIZE
+	default 98304 if WIFI
+	default 16384 if BT
+	default 4096
+
+choice BT_HCI_BUS_TYPE
+	default BT_ESP32 if BT
+endchoice

--- a/boards/xtensa/esp_wrover_kit/Kconfig.defconfig
+++ b/boards/xtensa/esp_wrover_kit/Kconfig.defconfig
@@ -7,19 +7,14 @@ config BOARD
 	default "esp_wrover_kit"
 	depends on BOARD_ESP_WROVER_KIT
 
-if BT
-
 config HEAP_MEM_POOL_SIZE
-	default 16384
-
-config ENTROPY_GENERATOR
-	default y
+	default 98304 if WIFI
+	default 16384 if BT
+	default 4096
 
 choice BT_HCI_BUS_TYPE
-	default BT_ESP32
+	default BT_ESP32 if BT
 endchoice
-
-endif # BT
 
 config DISK_DRIVER_SDMMC
 	default y

--- a/boards/xtensa/heltec_wifi_lora32_v2/Kconfig.defconfig
+++ b/boards/xtensa/heltec_wifi_lora32_v2/Kconfig.defconfig
@@ -6,3 +6,15 @@
 config BOARD
 	default "heltec_wifi_lora32"
 	depends on BOARD_HELTEC_WIFI_LORA32
+
+config ENTROPY_GENERATOR
+	default y
+
+config HEAP_MEM_POOL_SIZE
+	default 98304 if WIFI
+	default 16384 if BT
+	default 4096
+
+choice BT_HCI_BUS_TYPE
+	default BT_ESP32 if BT
+endchoice

--- a/boards/xtensa/m5stickc_plus/Kconfig.defconfig
+++ b/boards/xtensa/m5stickc_plus/Kconfig.defconfig
@@ -7,19 +7,14 @@ config BOARD
 	default "m5stickc_plus"
 	depends on BOARD_M5STICKC_PLUS
 
-config ENTROPY_ESP32_RNG
-	default y if ENTROPY_GENERATOR
-
-if BT
-
-config HEAP_MEM_POOL_SIZE
-	default 16384
-
 config ENTROPY_GENERATOR
 	default y
 
-choice BT_HCI_BUS_TYPE
-	default BT_ESP32
-endchoice
+config HEAP_MEM_POOL_SIZE
+	default 98304 if WIFI
+	default 16384 if BT
+	default 4096
 
-endif # BT
+choice BT_HCI_BUS_TYPE
+	default BT_ESP32 if BT
+endchoice

--- a/boards/xtensa/odroid_go/Kconfig.defconfig
+++ b/boards/xtensa/odroid_go/Kconfig.defconfig
@@ -13,16 +13,14 @@ config DISK_DRIVER_SDMMC
 config SPI
 	default y if DISK_DRIVER_SDMMC
 
-if BT
-
-config HEAP_MEM_POOL_SIZE
-	default 16384
-
 config ENTROPY_GENERATOR
 	default y
 
-choice BT_HCI_BUS_TYPE
-	default BT_ESP32
-endchoice
+config HEAP_MEM_POOL_SIZE
+	default 98304 if WIFI
+	default 16384 if BT
+	default 4096
 
-endif # BT
+choice BT_HCI_BUS_TYPE
+	default BT_ESP32 if BT
+endchoice

--- a/boards/xtensa/olimex_esp32_evb/Kconfig.defconfig
+++ b/boards/xtensa/olimex_esp32_evb/Kconfig.defconfig
@@ -8,4 +8,16 @@ if BOARD_OLIMEX_ESP32_EVB
 config BOARD
 	default "olimex_esp32_evb"
 
+config ENTROPY_GENERATOR
+	default y
+
+config HEAP_MEM_POOL_SIZE
+	default 98304 if WIFI
+	default 16384 if BT
+	default 4096
+
+choice BT_HCI_BUS_TYPE
+	default BT_ESP32 if BT
+endchoice
+
 endif # BOARD_OLIMEX_ESP32_EVB

